block/block.md

```diff
@@ -1 +1,99 @@
 # Block
+
+## Abstract
+
+The block manager is a key component of full nodes that is responsible for block production or block syncing depending on the node type. Block syncing in this context includes retrieving the published blocks from the network, validating them to raise fraud proofs upon validation failure, and updating the state and storing the validated blocks. A full node invokes multiple of the block manager functionalities in parallel, such as:
+
+* block production (only for sequencer full nodes)
+* block publication to DA network
+* block retrieval from DA network
+* block retrieval from Blockstore (which retrieves blocks from the p2p network)
+* block syncing
+
+## Protocol/Component Description
+
+The block manager is initialized using several parameters as defined below:
+
+* `signing key`: used for signing the blocks after production
+* `config`: block manager configurations (see config options below)
+* `genesis`: initialize the block manager with genesis state
+* `localstore`: store blocks and states
```

Comment:
yes it is persistent as the default is disk storage when the db path is specified in the config.

Reply:
added an assumption for that.

Reply:
if you specify in the config, it will be persistent storage, otherwise in-memory only.

Reply:
I think the default should be persistent storage right? Otherwise, people will always have to start from scratch. Maybe a flag ephemeral for it to stay in memory. That's more of a devEx question. cc @jcstein @MSevey thoughts ?

Reply:
I think the default being persistent would make sense. for example, if I ran a Polaris rollup and had deployed smart contracts to it for my applications, I would expect that starting back up my local environment would persist the state of the EVM chain by default. wouldn't this cause issues though if the devnet were stopped and restarted?

it should still be possible to wipe the state, too, so that when a user decides they want to, they can wipe the old chain history and start fresh.

i'm curious to see what Sevey thinks here, too

Reply:
Sorry missed this ping. 

I agree that the default should be persistent storage. The reason being that all defaults should be production defaults. The expectation of a production node is that if it stops, it will restart with the previous data. 

Having a flag for testing/dev/in-memory etc then makes sense.

Reply:
for cosmos-sdk apps, the default is always persistent storage (meaning, the db_dir is set to `data` by default) and rollkit will simply inherit that.